internal/xds/xdsclient/xdsresource/filter_chain.go

```diff
@@ -680,7 +680,7 @@ func processNetworkFilters(filters []*v3listenerpb.Filter) (*FilterChain, error)
 					// server-side." - A36
 					// Can specify v3 here, as will never get to this function
 					// if v2.
-					routeU, err := generateRDSUpdateFromRouteConfiguration(hcm.GetRouteConfig())
+					routeU, err := generateRDSUpdateFromRouteConfiguration(nil, hcm.GetRouteConfig())
```

Comment:
It is fine to pass `nil` here since this is server side listener resource processing and here we don't have to worry about authority rewriting. We *might* have to plump the decode options to this at some point in the future if they are required for something else on the server-side, but for now, we are good.